### PR TITLE
docs: update docs on links

### PR DESCRIPTION
**Pull request by Shigma**
**Summary**

Updated obsolete documents with new features of links.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/docs/docs/zh/guide/markdown.md [hunk 1]
```diff
@@ -41,11 +28,19 @@
...
+[bar - three](../bar/three.html) <!-- 具体文件可以使用 .html 结尾 -->
+[bar - four](../bar/four.md) <!-- 也可以用 .md -->
 ```
 
+### 链接的重定向 <Badge text="1.0.0-alpha.37"/>
+
+VuePress 支持重定向到干净链接。如果一个链接 `/foo` 找不到，VuePress 会自行寻找一个可用的 `/foo/` 或 `/foo.html`。反过来，当 `/foo/` 或 `/foo.html` 中的一个找不到时，VuePress 也会尝试寻找另一个。借助这种特性，我们可以通过官方插件 [@vuepress/plugin-clean-urls](../plugin/official/plugin-clean-urls.md) 定制你的网站路径。
+
+::: tip 注意
+无论是否使用了 permalink 和 clean-urls 插件，你的相对路径都应该依赖于当前的文件结构来定义。在上面的例子中，即使你将 `/foo/one.md` 的路径设为了 `/foo/one/`，你依然应该通过 `./two.md` 来访问 `/foo/two.md`。
```
Comment (ulivz) [inline packages/docs/docs/zh/guide/markdown.md]: 这里的 clean-urls 建议加上你插件的 github 地址

File: packages/docs/docs/zh/guide/markdown.md [hunk 2]
```diff
@@ -41,11 +28,19 @@
 ``` md
 [Home](/) <!-- 跳转到根部的 README.md -->
 [foo](/foo/) <!-- 跳转到 foo 文件夹的 index.html -->
-[foo heading anchor](/foo/#heading) <!-- 跳转到 foo/index.html 的特定 anchor 位置 -->
-[foo - one](/foo/one.html) <!-- 具体文件可以使用 .html 结尾 -->
-[foo - two](/foo/two.md) <!-- 也可以用 .md -->
+[foo heading](./#heading) <!-- 跳转到 foo/index.html 的特定标题位置 -->
+[bar - three](../bar/three.html) <!-- 具体文件可以使用 .html 结尾 -->
+[bar - four](../bar/four.md) <!-- 也可以用 .md -->
```
Comment (ulivz) [inline packages/docs/docs/zh/guide/markdown.md]: 建议优先推荐用户使用 `.md`, 有利于 jump-to-target.

File: packages/docs/docs/zh/guide/markdown.md (hunk 1 above)
Comment (Shigma) [reply packages/docs/docs/zh/guide/markdown.md]: 上一段中已经有链接了，所以认为这里不需要了。

File: packages/docs/docs/zh/guide/markdown.md [hunk 3]
```diff
@@ -41,11 +28,19 @@
 ``` md
```
Comment (ulivz) [inline packages/docs/docs/zh/guide/markdown.md]: > 假设你现在在 `foo/one.md` 中

上面的这句话建议移到这个的前面，我作为阅读者找了一会儿才看到这个前提。

Comment (Shigma) [reply packages/docs/docs/zh/guide/markdown.md]: 已修改~

File: packages/docs/docs/zh/guide/markdown.md (hunk 1 above)
Comment (ulivz) [reply packages/docs/docs/zh/guide/markdown.md]: 恩恩~

**Pull request merged**